File: estudos-pf/nivel-1-map-filter-reduce/solution/q1/q1.py
def multiplo_3(valor: int) -> bool:
    return valor % 3 == 0

def soma_multiplos_de_3(sequencia: list) -> int:
    return sum(filter(multiplo_3, sequencia)) 

assert soma_multiplos_de_3([]) == 0
assert soma_multiplos_de_3([1, 2, 4, 5]) == 0
assert soma_multiplos_de_3([3, 6, 9]) == 18
assert soma_multiplos_de_3([1, 3, 5, 9]) == 12
assert soma_multiplos_de_3([-3, 3]) == 0

File: estudos-pf/nivel-1-map-filter-reduce/solution/q1/q1.clj
(defn multiplo3? [valor]
  (= (mod valor 3) 0))

(defn soma-multiplo-3 [sequencia]
  (reduce + (filter multiplo3? sequencia)))

(assert (= (soma-multiplo-3 []) 0))
(assert (= (soma-multiplo-3 [1 2 4 5]) 0))
(assert (= (soma-multiplo-3 [3 6 9]) 18))
(assert (= (soma-multiplo-3 [1 3 5 9]) 12))
(assert (= (soma-multiplo-3 [-3 3]) 0))
